File: Numerical Analysis/Root Of High Degree Equation/BisectionMethod.py
y = lambda x: 2*x**2 - 5*x + 3
x1 = float(input('Enter the value x1 : '))
x2 = float(input('Enter the value x2 : '))
y1 = y(x1)
y2 = y(x2)
if y1*y2 > 0:
	print('The root is not within the given interval')
	exit
for bisection in range(1,101):
	xh = (x1 + x2)/2
	yh = y(xh)
	y1 = y(x1)
	if abs(y1) < 1.0E-6:
		break
	elif y1 * yh <0:
		x2 = xh
	else:
		x1 = xh
print('The root : %0.5f' % x1)
print('The number of bisections : %d' % bisection)

File: BisectionMethod.m
b = @(x) 2*x^2 - 5*x + 3;
low=input('Enter the value of lower range : '); 
high=input('Enter the value of higher range : ');
tol=input('Enter the  value : ');
if b(low)==0
  fprintf('Lower value is the root\n');
  return
elseif b(high)==0
  fprintf('Higher value is the root\n');
  return
end

if b(high)*b(low)>0
  fprintf('No root exist in that interval\n');
 return 
end


x1=(low+high)/2;
i=1;
fprintf('\n');
while abs(b(x1))>=tol
  x1=(low+high)/2;
  fprintf('  %d\t  %f\t%f\t%f\t%f\n',i,low,high,x1,b(x1));
  i=i+1;
  if b(low)*b(x1)>0
    low=x1;
  else
    high=x1;
  endif
endwhile

